jerry-debugger/src/lib/debugger-client.ts

```diff
@@ -0,0 +1,404 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import assert from 'assert';
+import WebSocket from 'ws';
+
+import { Breakpoint, ParsedFunction } from './breakpoint';
+import { PROTOCOL } from './jrs-protocol-constants';
+import { FunctionMap, ParsedSource } from './parsed-source';
+import { cesu8ToString } from './string-utils';
+
+export interface JerryDebuggerOptions {
+  host?: string;
+  port?: number;
+  verbose?: boolean;
+}
+
+export const DEFAULT_DEBUGGER_HOST = 'localhost';
+export const DEFAULT_DEBUGGER_PORT = 5001;
+
+// expected JerryScript debugger protocol version
+const JERRY_DEBUGGER_VERSION = 1;
+
+export class JerryDebuggerClient {
+  readonly host: string;
+  readonly port: number;
+  readonly verbose: boolean;
+  private socket?: WebSocket;
+
+  // debugger configuration
+  private maxMessageSize: number = 0;
+  private cpointerSize: number = 0;
+  private littleEndian: boolean = true;
+  private version: number = 0;
+  private parseObj?: ParsedSource;
+  private outputResult?: Uint8Array;
+  private exceptionData?: Uint8Array;
+  private lastBreakpointHit?: object;
+  private backtraceFrame: number = 0;
+  functions: FunctionMap = {};
+
+  constructor(options: JerryDebuggerOptions) {
+    this.host = options.host || DEFAULT_DEBUGGER_HOST;
+    this.port = options.port || DEFAULT_DEBUGGER_PORT;
+    this.verbose = options.verbose || false;
+  }
+
+  connect() {
+    if (this.socket) {
+      this.disconnect();
```

Comment:
Still think it makes more sense to `return Promise.resolve();` here instead of disconnecting.